### PR TITLE
Feature/normalize errors

### PR DESCRIPTION
**Pull request by nick13jaremek**

### CODE REVIEW

File: lib/platforms/identity.js [hunk 1]
```diff
@@ -162,7 +158,7 @@ function createIdentity(identityObj, callback) {
       subscribeIdentityToChannels(savedIdentity.id, savedIdentity.channels, function(err, subscribedIdentity) {
 
         if (err) {
-          return done(err);
+          return done(err); // Controlled error
```
Comment (guumaster) [inline lib/platforms/identity.js]: remove unnecessary comments

File: lib/middleware/validate_identity.js [hunk 2]
```diff
@@ -35,7 +42,11 @@ module.exports = function() {
       if (!identity) {
 
         log.error('Identity not found');
-        return next(new errors.UnauthorizedError('Identity not found'));
+        err = new errors.UnauthorizedError('Identity not found');
```
Comment (guumaster) [inline lib/middleware/validate_identity.js]: why this is not ` return next(new errors.UnauthorizedError('Identity not found'));` ?

File: lib/platforms/identity.js [hunk 3]
```diff
@@ -288,7 +288,7 @@ function findIdentitiesByFieldValue(field, values, callback) {
   Identity.find().where(field).in(values).exec(function(err, foundIdentities) {
 
     if (err) {
-      return callback(err);
+      return callback(new errors.InternalError('Could not find identities for field ' + field + ' and values ' + values));
```
Comment (guumaster) [inline lib/platforms/identity.js]: Not found stuff should return 404, not 500.

**Comment by guumaster** [comment]
Wrong ids on searches should not return `InternalError`, but 404 or 400.

File: lib/platforms/identity.js (hunk 3 above)
Comment (nick13jaremek) [inline lib/platforms/identity.js]: @guumaster  Perhaps the messages are incorrect. When a database error occurs it does not necessary mean that the requested object(s) were not found, but it means that something bad happened while attempting to manipulate them. The NotFoundError should apply when the returned results are empty or undefined.

Therefore, I can change the messages to something like `Fatal error while searching for Identity`, and keep the 500 status.

**Comment by luismesas** [comment]
guumaster, sorry I accepted, was ok or need to reverse the merge?

**Comment by nick13jaremek** [comment]
I have most of the fixes commented here ready, the only remaining point would be knowing an answer to my proposition for 500->400/404 error status issue.

**Comment by guumaster** [comment]
@luismesas,  that's ok.  @nick13jaremek, if you change the error to a more general message, at least log the original error to know what the problem is. 

A way to do that is leave the error as is until the final error handler. If it is a uncommon error, log it and then change it to "An Internal error occurred" for the user. That way you don't have to write lots of InternalError all over.

**Comment by nick13jaremek** [comment]
@guumaster Okay, so it would be the same as returning the uncommon error as it is up to the error handler, which would return to the user a 500 InternalError with a custom message or whatever message is contained in the error.

Uncommon errors will be of *Error* type, so the error handler will transform them into a 500 HttpError.

**Comment by guumaster** [comment]
:+1: exactly. Awful Error => Log all the things => Nice HttpError 500 :facepunch:

**Comment by nick13jaremek** [comment]
Okay, will apply those changes and create a new pull request.

**Pull request merged**